dashboard/src/components/error-boundary.tsx

```diff
@@ -0,0 +1,159 @@
+'use client';
+
+import React, { useCallback } from 'react';
+import { AlertTriangle, RefreshCw, Home, Mail, ArrowLeft } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import { Card, CardContent } from '@/components/ui/card';
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+  error?: Error;
+  errorInfo?: React.ErrorInfo;
+}
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+  fallback?: React.ComponentType<{ error?: Error; resetError: () => void }>;
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Comment:
```suggestion
export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Whats the purpose of using the extend syntax here, instead of writing it as a function?